Framework/Crystal/src/AnvredCorrection.cpp

```diff
@@ -114,6 +116,22 @@ void AnvredCorrection::init() {
   defineProperties();
 }
 
+std::map<std::string, std::string> AnvredCorrection::validateInputs() {
+  std::map<std::string, std::string> result;
+
+  const double radius = getProperty("Radius");
+  if (radius == EMPTY_DBL()) {
+    // check that if radius isn't supplied that the radius can be accessed from the sample
+    m_inputWS = getProperty("InputWorkspace");
+    const auto &sampleShape = m_inputWS->sample().getShape();
+    if (!sampleShape.hasValidShape() ||
+        sampleShape.shapeInfo().shape() != Geometry::detail::ShapeInfo::GeometryShape::SPHERE) {
```

Comment:
Does this need to be a sphere or would this work with cylinders too as per the [implementation](https://doxygen.mantidproject.org/nightly/d0/d72/ShapeInfo_8cpp_source.html)?

Secondly, do we need to check that `shapeInfo().radius != 0`?

Reply:
The implementation does not work for cylinders - or rather cylinders require different coefficients for the polynomial in the array `pc`.
I just checked and it is possible to set a sample to have a radius of 0, in this case `mur = mu*radius = 0` - this is fine for the code, which will evaluate the absorption correction to be `astar=1`. Perhaps more worrying is that I've just checked and I can set a negative radius!
A negative radius will at least throw an error later on
https://github.com/mantidproject/mantid/blob/e3b55babd91d978685a38e84605b957ed823ab04/Framework/Crystal/src/AnvredCorrection.cpp#L432-L433
But I can add some validation if you like?

Reply:
Perhaps there should be some validation on the shape itself to ensure radius is not negative?

Reply:
Is it possible to set a negative radius using setSample?
If so, I would move it to validation. Otherwise, it implies a broken IDF / something screwy going on and we can leave the check to last minute as it's not easy for a user to accidentally do

Reply:
It isn't possible to set a negative radius when the user supplies it as an argument to the algorithm (we validate it is always positive). but it is possible for the user to set a sample shape as a sphere with a negative radius as so.
```
shape = '''<sphere id="V-sphere">
    <centre x="0.0"  y="0.0" z="0.0" />
    <radius val="0.0025"/>
    </sphere>'''
SetSample(ws, Geometry={'Shape': 'CSG', 'Value': shape})
```

Reply:
> > I think this is actually because muR is too large - rather than the sample radius being 0
> 
> Ah I didn't check correctly if `CreateSampleWorkspace` assigned a sample shape / radius that was spherical. Can we emit this warning once as it completely fill the log and wipes out all other info? Also the `std::endl` should be replaced with `\n` as it forces a buffer flush in the logging layer

The trouble is the function which contains this check, `absor_sphere`, is called in a loop over events/bins, but I could print a single warning outrside the loop over events (probably do one per spectrum) if I add a bool `invalid_muR` argument to `absor_sphere` - would that work for you?